sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -2138,7 +2138,12 @@ def predict(self, X):
             The predicted classes.
         """
         # TODO: This could be done in parallel
-        encoded_classes = np.argmax(self.predict_proba(X), axis=1)
+        raw_predictions = self._raw_predict(X)
+        if raw_predictions.shape[1] == 1:
+            # np.argmax([0, 0]) is 0, not 1, therefor "> 0" not ">= 0"
+            encoded_classes = (raw_predictions.ravel() > 0).astype(np.int64)
```

Comment:
The [CI failure](https://dev.azure.com/scikit-learn/scikit-learn/_build/results?buildId=61376&view=logs&j=aabdcdc3-bb64-5414-b357-ed024fe8659e&t=b7b3ba55-d585-563b-a032-f235636c22b0) looks real.

I suspect this works:

```suggestion
            encoded_classes = (raw_predictions.ravel() > 0).astype(int)
```

(or `np.intp`)